src/actions/posts.js

```diff
@@ -899,8 +908,8 @@ export function getNeededCustomEmojis(state, posts) {
     return customEmojisToLoad;
 }
 
-export function removePost(post) {
-    return (dispatch, getState) => {
+export function removePost(post: Post): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
```

Comment:
Why change this to be an async function?

Reply:
I asked earlier in the developer community what some thought as to if we should standardize them all to be the same since they all have the behavior of returning a dispatch and getstate function and it seemed that this was the preference.

Reply:
Oh, I see. I didn't realize that the `ActionFunc` required it to be async. This is fine to leave as is then, although I'm generally not a fan of adding async everywhere since it makes it harder to test this code.

Reply:
I didn't realise that we are changing them to be promises for `ActionFunc` type when this was being discussed. Rather than changing this to be `async` i think we should change the type to be either a `func` or a `promise`. Same with `return`. If we are returning `{data: true}` just for the sake of returning  then it does not add much value so we probably should change the type for return too. 
i am 0/5 on the change for return type but for changing async i am at 2/5 that we should change the type to be forgiving if it is a a simple func.

@jespino Thoughts?

Reply:
Yes, I think is a valid usage to have sync actions, so I agree with @sudheerDev, we should change the `ActionFunc` to accept both, regular functions or async functions. about the return, I would say was the pre-existing pattern, so I wouldn't change it for now, maybe in the future, we have to rethink a lot of things in mattermost-redux.

Reply:
@Azero123 Can you change the type of `ActionFunc` to accept a func? And change back the instances of newly added async function to be sync as before?. I am good with changing return type to be `{data: true}`.
Sorry about the confusion and delay on reviewing this.